Classes/Service/OAuthService.php

```diff
@@ -251,20 +251,20 @@ public function getFreshAccessToken(): ?AccessToken
         }
         $accessToken = new AccessToken($options);
 
-        if ($accessToken->hasExpired()) {
-            try {
-                $newAccessToken = $this->getProvider()->getAccessToken(new RefreshToken(), [
-                    'refresh_token' => $accessToken->getRefreshToken(),
-                ]);
-
-                // TODO
-            } catch (IdentityProviderException $e) {
-                // TODO: log problem
-                return null;
-            }
+        if ($accessToken->hasExpired() === false) {
```

Comment:
since hasExpired returns bool already, there is no need for an additional comparison.
We can use normal bool operators. => `!$accessToken->hasExpired()`

Reply:
Nope, not a matter of taste, matter of how things work. ;-)
"no need to compare a bool to a static bool"

`!` means negate the bool and continue (boils down to 1 CPU instruction "conditional jmp")

`=== false` means extra CPU cycles to check for matching data type and then to compare two bool values.
(if a compiler is good, it can reduce this to the above, but I doubt that in the case of PHP)

Anyway: Thanks for fixing.